ee/tabby-schema/src/schema/web_documents.rs

```diff
@@ -35,15 +36,23 @@ impl CustomWebDocument {
 
 #[derive(Validate, GraphQLInputObject)]
 pub struct CreateCustomDocumentInput {
-    #[validate(url(code = "name", message = "Invalid Name"))]
+    #[validate(regex(
+        code = "name",
+        path = "*crate::schema::constants::USERNAME_REGEX",
+        message = "Invalid repository name"
+    ))]
     pub name: String,
     #[validate(url(code = "url", message = "Invalid URL"))]
     pub url: String,
 }
 
 #[derive(Validate, GraphQLInputObject)]
 pub struct SetPresetDocumentActiveInput {
-    #[validate(url(code = "name", message = "Invalid Name"))]
+    #[validate(regex(
+        code = "name",
+        path = "*crate::schema::constants::USERNAME_REGEX",
```

Comment:
let's create a separate regex for web doc name (e.g starts with [a-zA-Z], can only contains space and [a-zA-Z0-9]

Reply:
Update